pad_ui/PackDownloadDiv.gd

```diff
@@ -0,0 +1,151 @@
+"""
+	Copyright 2020 Google LLC
+	
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+	
+		https://www.apache.org/licenses/LICENSE-2.0
+	
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+"""
+"""
+	Object used to display one row of UI for handling asset pack downloads
+"""
+
+extends Node2D
+signal fetched_pack(pack_name)
+signal removed_pack(pack_name)
+
+var request_obj : PlayAssetPackFetchRequest
+var pack_name : String
+var pad_manager : PlayAssetPackManager
+
+func init(_pack_name : String, _pad_manager : PlayAssetPackManager):
+	pack_name = _pack_name
+	pad_manager = _pad_manager
+	$PackName.text = pack_name
+	# get if it is installed or not
+	var pack_installed = pad_manager.get_pack_location(pack_name) != null
+	if pack_installed:
+		_show_completed_ui()
+	else:
+		_show_init_ui()
+
+func _convert_byte_to_stepified_megabytes(byte_value : int):
+	# convert bytes to MB with 2 digits of precision
+	return stepify(byte_value / 1000000, 0.01)
+
+func _process(delta):
+	if request_obj == null:
+		return
+
+	var should_update_ui = request_obj.get_state().get_status() != PlayAssetPackManager.AssetPackStatus.UNKNOWN
+	
+	if should_update_ui:
+		# update the downloading UI
+		var bytes_downloaded = float(request_obj.get_state().get_bytes_downloaded())
+		var bytes_to_download = float(request_obj.get_state().get_total_bytes_to_download())
+		var progress_ratio = bytes_downloaded / bytes_to_download
```

Comment:
The only purpose of `progress_ratio` is to convert it to an int below. Could do this entire section with int only arithmetic, i.e. by multiplying by `100` first.

(See related comment below.)

Reply:
I think floating point division is required since if we are doing int only division, it will only return [0 or 1](https://godotengine.org/qa/4588/1-4-0-not-25-why-isnt-division-working-in-gdscript)
I refactored this part by updating the range of the ProgressBar to 0-1 (from 0-100), so that we can directly pass this ```progress_ratio``` to the ProgressBar.